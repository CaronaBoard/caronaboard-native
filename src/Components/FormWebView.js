import React, { Component } from 'react'
import { WebView } from 'react-native'

export default class FormWebView extends Component {
  render () {
    return (
      <WebView source={{uri: 'https://google.com'}} />
    )
  }
}
