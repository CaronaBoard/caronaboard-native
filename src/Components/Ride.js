/* @flow */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Text, TouchableOpacity } from 'react-native'
import Styles from './Styles/RideStyles'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { RkCard } from 'react-native-ui-kitten'

export default class Ride extends Component {
  renderLine (subTitle, icon) {
    return (
      <View style={Styles.line} >
        <Icon name={icon} style={Styles.icon} />
        <Text rkCardText>{subTitle}</Text>
      </View>
    )
  }

  renderFlexible (flexible) {
    return flexible ? this.renderLine('Rota flex√≠vel', 'call-split') : <View />
  }

  render () {
    const {
        area,
        origin,
        destination,
        flexible,
        days,
        hours,
        name,
        formUrl
     } = this.props.ride

    return (
      <TouchableOpacity onPress={() => console.log('touched ' + formUrl)}>
        <View>
          <RkCard>
            <View rkCardContent>
              <Text rkCardTitle>{area}</Text>
              { this.renderLine(origin, 'radio-button-unchecked') }
              { this.renderLine(destination, 'radio-button-unchecked') }
              { this.renderFlexible(flexible) }
            </View>
            <View rkCardContent>
              { this.renderLine(days, 'today') }
              { this.renderLine(hours, 'schedule') }
              { this.renderLine(name, 'directions-car') }
            </View>
          </RkCard>
        </View>
      </TouchableOpacity>
    )
  }
}

Ride.propTypes = {
  ride: PropTypes.shape({
    area: PropTypes.string.isRequired,
    days: PropTypes.string.isRequired,
    destination: PropTypes.string.isRequired,
    formUrl: PropTypes.string.isRequired,
    hours: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    origin: PropTypes.string.isRequired,
    flexible: PropTypes.bool.isRequired
  }).isRequired
}
