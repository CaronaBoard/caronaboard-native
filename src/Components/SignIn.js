/* @flow */

import React, { Component } from 'react'
import {
  View,
  Text,
  TextInput
} from 'react-native'

export default class MyComponent extends Component {
  constructor () {
    super()
    this.state = {text: ''}
  }

  render () {
    return (
      <View>
        <Text>{'Email'}</Text>
        <TextInput
          onChangeText={(text) => this.setState({text})}
          value={this.state.text} />
        <Text>{'Password'}</Text>
        <TextInput />
      </View>
    )
  }
}
