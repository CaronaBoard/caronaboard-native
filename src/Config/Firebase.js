import Firebase from 'firebase'

const config = {
  apiKey: 'AIzaSyDfwHLwWKTqduazsf4kjbstJEA2E1sCeoI',
  authDomain: 'caronaboard-61f75.firebaseapp.com',
  databaseURL: 'https://caronaboard-61f75.firebaseio.com',
  storageBucket: 'caronaboard-61f75.appspot.com',
  messagingSenderId: '617045704123'
}

const setup = () => {
  console.log('firebase setup ' + Firebase)
  return Firebase.initializeApp(config)
}

export default { setup }
