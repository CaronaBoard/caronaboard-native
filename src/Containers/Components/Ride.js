/* @flow */

import React, { Component } from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

export default class Ride extends Component {
  render () {
    const {
        area,
        days,
        destination,
        flexible,
        formUrl,
        hours,
        name,
        origin
     } = this.props.ride

    return (
      <View style={styles.container}>
        <Text>{area}</Text>
        <Text>{days}</Text>
        <Text>{destination}</Text>
        <Text>{flexible}</Text>
        <Text>{formUrl}</Text>
        <Text>{hours}</Text>
        <Text>{name}</Text>
        <Text>{origin}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  }
})
