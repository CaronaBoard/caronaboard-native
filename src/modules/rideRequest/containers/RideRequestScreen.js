import React, { Component } from 'react'
import { View } from 'react-native'

import { GradientButton } from '../../shared/components'
import { Ride, ridePropTypes } from '../components/Ride'
import { saveRideRequest } from '../../../services/firebase'
import styles from './styles/RideRequestScreenStyles'

export class RideOfferScreen extends Component {
  render () {
    const { ride } = this.props
    return (
      <View style={styles.flexible}>
        <View style={styles.container}>
          <View style={styles.inputTextsContainer}>
            <Ride ride={ride} />
          </View>
        </View>
        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            text={'SAVE'}
            onPress={this.onButtonSubmit}
          />
        </View>
        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            text={'SAVE'}
            onPress={() => saveRideRequest(ride.rideId)}
          />
        </View>
      </View>
    )
  }
}

RideOfferScreen.propTypes = ridePropTypes

export default RideOfferScreen
