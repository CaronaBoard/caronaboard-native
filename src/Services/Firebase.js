import Firebase from 'firebase'
import _ from 'lodash'

const config = {
  apiKey: 'AIzaSyDfwHLwWKTqduazsf4kjbstJEA2E1sCeoI',
  authDomain: 'caronaboard-61f75.firebaseapp.com',
  databaseURL: 'https://caronaboard-61f75.firebaseio.com',
  storageBucket: 'caronaboard-61f75.appspot.com',
  messagingSenderId: '617045704123'
}

let database = (() => {
  Firebase.initializeApp(config)
  return Firebase.database().ref()
})()

const toArrayOfRides = (firebaseResponse) => {
  return _.flatMap(firebaseResponse, Object.values)
}

export const getAllRides = () => {
  return new Promise(resolve => {
    database.child('rides').once('value').then(
      (snapshot) => resolve(toArrayOfRides(snapshot.val()))
    )
  })
}
