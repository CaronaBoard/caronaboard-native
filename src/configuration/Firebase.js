import Firebase from 'firebase'

const config = {
  apiKey: 'AIzaSyDfwHLwWKTqduazsf4kjbstJEA2E1sCeoI',
  authDomain: 'caronaboard-61f75.firebaseapp.com',
  databaseURL: 'https://caronaboard-61f75.firebaseio.com',
  storageBucket: 'caronaboard-61f75.appspot.com',
  messagingSenderId: '617045704123'
}

const initialize = () => {
  Firebase.initializeApp(config)
}

export default { initialize }
