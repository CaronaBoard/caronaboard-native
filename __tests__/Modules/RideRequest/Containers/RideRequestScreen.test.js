import React from 'react'
import { shallow } from 'enzyme'
import { RideOfferScreen } from '../../../../src/Modules/RideRequest/Containers/RideRequestScreen'
import { Button } from '../../../../src/Modules/Shared/Components'

import * as Firebase from '../../../../src/Services/Firebase'
jest.mock('../../../../src/Services/Firebase')

const props = {
  ride: {
    'driverId': 'driverId',
    'rideId': 'rideId',
    'area': 'Aeroporto',
    'days': 'Seg a Sex',
    'destination': 'Rodoviaria',
    'hours': '19h',
    'origin': 'Tecnopuc',
    'profile': {
      'contact': {
        'kind': 'Whatsapp', 'value': '5198269999'
      },
      'name': 'Eduardo' }
  }
}

describe.only('<RideOfferScreen />', () => {
  const wrapper = shallow(<RideOfferScreen {...props} />)

  it('Snapshot have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call firebase service to request a ride', () => {
    // TODO: I Think that' not a good idea calling services directly, we should use action
    Firebase.saveRideRequest = jest.fn()
    wrapper.find(Button).simulate('press')
    expect(Firebase.saveRideRequest).toHaveBeenCalledWith(props.ride.rideId)
  })
})
