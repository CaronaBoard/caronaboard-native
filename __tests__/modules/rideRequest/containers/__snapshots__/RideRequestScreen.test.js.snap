// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RideOfferScreen /> Snapshot have a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
        "paddingHorizontal": 50,
      }
    }
  >
    <View
      style={
        Object {
          "minHeight": 370,
        }
      }
    >
      <Ride
        ride={
          Object {
            "area": "Aeroporto",
            "days": "Seg a Sex",
            "destination": "Rodoviaria",
            "driverId": "driverId",
            "hours": "19h",
            "origin": "Tecnopuc",
            "profile": Object {
              "contact": Object {
                "kind": "Whatsapp",
                "value": "5198269999",
              },
              "name": "Eduardo",
            },
            "rideId": "rideId",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <GradientButton
      rkType="large"
      text="SAVE"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <GradientButton
      onPress={[Function]}
      rkType="large"
      text="SAVE"
    />
  </View>
</View>
`;
